monkey/monkey_island/cc/ui/src/components/report-components/SecurityReport.js

```diff
@@ -226,6 +226,18 @@ class ReportPageComponent extends AuthComponent {
     return Object.keys(this.state.report).length === 0;
   }
 
+  getMonkeyDuration() {
+    if (this.state.report.overview.monkey_duration !== 0) {
```

Comment:
We don't need to check the duration, just call `didAllAgentsShutdown`

Reply:
The promise doesn't resolve in time and I don't think we should crowd this with state for everything when we're going to rework it anyway. I'm leaving it as it is.

Reply:
I'd anticipate that report will need to show the duration even after the rework. Yes, that method queries the island even though we already have the list of agent. We should separate the request part from parsing the agent list part.